### PR TITLE
Rebase with gitenterprise repo(till PR93) + aiebu-asm static building

### PR DESCRIPTION
**Pull request by HimanshuChoudhary-Xilinx**
1. Rebase with gitenterprise repo(till PR93) 
2. Enabled aiebu-asm static building for OS which openssl_static(libcrypto.a) libs installed/supported like ubuntu/centos
    and dynamic linking for OS **not** having openssl_static(libcrypto.a) libs  installed/supported like Redhat9   
3. Built in XRT pipeline https://github.com/Xilinx/XRT/actions/runs/11360566026?pr=8530

### CODE REVIEW

File: src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp [hunk 1]
```diff
@@ -336,8 +339,11 @@ namespace aiebu {
           auto bw_header = reinterpret_cast<const XAie_BlockWrite32Hdr_opt *>(ptr);
           auto payload = reinterpret_cast<const char*>(ptr + sizeof(XAie_BlockWrite32Hdr_opt));
           auto offset = static_cast<uint32_t>(payload-mc_code.data());
-          uint64_t buffer_length_in_bytes = reinterpret_cast<const uint32_t*>(payload)[0] * 4;
-          blockWriteRegOffsetMap[bw_header->RegOff] = std::make_pair(offset, buffer_length_in_bytes);
+          uint32_t size = (bw_header->Size - sizeof(*bw_header));
```
Comment (sonals) [inline src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp:342]: Can you add a few lines of comment to describe the operation here?

File: src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp [hunk 2]
```diff
@@ -252,8 +252,11 @@ namespace aiebu {
           auto bw_header = reinterpret_cast<const XAie_BlockWrite32Hdr *>(ptr);
           auto payload = reinterpret_cast<const char*>(ptr + sizeof(XAie_BlockWrite32Hdr));
           auto offset = static_cast<uint32_t>(payload-mc_code.data());
-          uint64_t buffer_length_in_bytes = reinterpret_cast<const uint64_t*>(payload)[0] * 4;
-          blockWriteRegOffsetMap[bw_header->RegOff] = std::make_pair(offset, buffer_length_in_bytes);
+          uint32_t size = (bw_header->Size - sizeof(*bw_header));
+          for (auto bd = 0U ; bd < size; bd+=SHIM_DMA_BD_SIZE) { //size and bd in bytes
```
Comment (sonals) [inline src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp:256]: Can you add a few lines of comment to describe the operation here?

**Review by sonals — CHANGES_REQUESTED** [review]
Please make the code documentation changes and then we should merge it.

File: src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp (hunk 1 above)
Comment (HimanshuChoudhary-Xilinx) [reply src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp:346]: done

File: src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp (hunk 2 above)
Comment (HimanshuChoudhary-Xilinx) [reply src/cpp/aiebu/src/preprocessor/aie2/aie2_blob_preprocessor_input.cpp:258]: added

**Review by sonals — APPROVED** [review]

**Pull request merged**